From f733b3a93eef843bfeae6f241a4f6fac6b95d656 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:09:03 -0400
Subject: [powerpc] Fix xmon disassembler for little-endian

Message-id: <1410545655-205645-315-git-send-email-dzickus@redhat.com>
Patchwork-id: 94257
O-Subject: [RHEL7 PATCH 314/626] powerpc: Fix xmon disassembler for little-endian
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 72eceef67abbe596a4e93ee79e08d9e6c35430ae
Author: Philippe Bergheaud <felix@linux.vnet.ibm.com>
Date:   Mon Dec 2 10:10:12 2013 +0100

    powerpc: Fix xmon disassembler for little-endian

    This patch fixes the disassembler of the powerpc kernel debugger xmon,
    for little-endian.

    Signed-off-by: Philippe Bergheaud <felix@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/xmon/xmon.c b/arch/powerpc/xmon/xmon.c
index b079098..08504e7 100644
--- a/arch/powerpc/xmon/xmon.c
+++ b/arch/powerpc/xmon/xmon.c
@@ -171,7 +171,11 @@ extern void xmon_leave(void);
 #define REG  "%.8lx"
 #endif
 
+#ifdef __LITTLE_ENDIAN__
+#define GETWORD(v) (((v)[3] << 24) + ((v)[2] << 16) + ((v)[1] << 8) + (v)[0])
+#else
 #define GETWORD(v) (((v)[0] << 24) + ((v)[1] << 16) + ((v)[2] << 8) + (v)[3])
+#endif
 
 #define isxdigit(c) (('0' <= (c) && (c) <= '9') \
     || ('a' <= (c) && (c) <= 'f') \
-- 
1.7.1